src/main/scala-2.11/org/apache/james/gatling/control/UserFeeder.scala

```diff
@@ -1,30 +1,68 @@
 package org.apache.james.gatling.control
 
-import java.util.UUID
-
 import io.gatling.core.Predef._
+import org.apache.james.gatling.utils.RandomStringGenerator
 
 import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.Future
+import scala.concurrent.duration.Duration.Inf
+import scala.concurrent.{Await, Future}
+import scala.util.Success
 
 object UserFeeder {
 
-  def createUserFeeder(userCount: Int): Array[Map[String, String]] = {
-    val domain = Domain(UUID.randomUUID().toString)
+  val USERNAME: String = "username"
+  val PASSWORD: String = "password"
+
+  def createUserFeeder(userCount: Int): Array[Map[String, String]] =
+    toFeeder(
+      awaitInitialization(
+        createRegisteredUserFutures(userCount)))
+
+  def createUserFeederWithInboxAndOutbox(userCount: Int): Array[Map[String, String]] =
+    toFeeder(
+      awaitInitialization(
+        createRegisteredUserFutures(userCount)
+          .map(addSystemMailboxes(_))))
 
+  private def awaitInitialization(userFutures: List[Future[User]]) =
+    Await.result(
+      Future.sequence(userFutures),
+      Inf)
+
+  private def createRegisteredUserFutures(userCount: Int): List[Future[User]] = {
+    val domain = Domain()
     JamesWebAdministration.addDomain(domain).get
 
-    val users = (0 until userCount)
-      .map(i => User(
-        Username(s"""${UUID.randomUUID().toString}@${domain.value}"""),
-        Password(UUID.randomUUID().toString)))
+    generateUsers(userCount, domain)
+      .map(registerUser(_))
+  }
+
+  private def generateUsers(userCount: Int, domain: Domain): List[User] =
+    (0 until userCount)
+      .map(i => createRandomUser(domain))
       .toList
 
-    Future.sequence(
-      users.map(user => JamesWebAdministration.addUser(user)))
-      .get
+  private def createRandomUser(domain: Domain): User = {
+    val localAddressPart = RandomStringGenerator.randomString
+    User(Username(s"$localAddressPart@${domain.value}"),
+      Password())
+  }
 
-    users.map(user => Map("username" -> user.username.value, "password" -> user.password.value)).toArray
+  private def registerUser(user: User): Future[User] = {
+      JamesWebAdministration.addUser(user)
+        .map(response => user)
   }
 
+  private def addSystemMailboxes(userFuture: Future[User]): Future[User] =
+    userFuture.andThen {
```

Comment:
instead of chaining the call, you should return the function to apply and do the chaining at an upper level